rust/benchmarks/src/bin/tpch.rs

```diff
@@ -400,7 +400,7 @@ fn create_logical_plan(ctx: &mut ExecutionContext, query: usize) -> Result<Logic
                             (n1.n_name = 'FRANCE' and n2.n_name = 'GERMANY')
                             or (n1.n_name = 'GERMANY' and n2.n_name = 'FRANCE')
                         )
-                        and l_shipdate > date '1995-01-01' and l_shipdate < date '1996-12-31'
+                        and l_shipdate between date '1995-01-01' and date '1996-12-31'
```

Comment:
I think we can use the TPC-H Query validation section to resolve the use of `<` or `<=` etc by running the query against SF1 data and comparing the result

According to the spec, the expected answer is `123141078.23` at SF1 

http://www.tpc.org/tpc_documents_current_versions/pdf/tpc-h_v2.17.2.pdf
page 38


```
REVENUE
123141078.23
```

Reply:
That's a great idea, I agree this would be very valuable.

Reply:
I am working with Andy to increase the testing provided by the TPC-H queries here: https://github.com/apache/arrow/pull/8760#issuecomment-744076299

The plan is to verify results against the correct results when run in test mode.

Reply:
When implementing BETWEEN I went hunting for the ANSI SQL spec but the best I could do is find the Postgres implementation which confirms it is definitely INCLUSIVE.

https://www.postgresql.org/docs/13/functions-comparison.html

```
The BETWEEN predicate simplifies range tests:

a BETWEEN x AND y
is equivalent to

a >= x AND a <= y
```